sys/include/net/loramac.h

```diff
@@ -114,233 +98,410 @@ extern "C" {
 
 /**
  * @brief   Default device class (A, B or C)
+ *
+ * Configure the class of device. LoRaWAN supports three classes of operation
+ * for end nodes namely Class A, B and C. All LoRaWAN devices are expected to
+ * implement Class A, whereas Class B and Class C can be considered as
+ * extensions to the specification of Class A devices.
+ *
+ * @see     [LoRaWAN Classes by TTN](https://www.thethingsnetwork.org/docs/lorawan/classes.html)
+ *
+ * @note    GNRC LoRaWAN only supports Class A
  */
-#ifndef LORAMAC_DEFAULT_DEVICE_CLASS
-#define LORAMAC_DEFAULT_DEVICE_CLASS           (LORAMAC_CLASS_A)
+#if IS_ACTIVE(CONFIG_LORAMAC_DEFAULT_DEVICE_CLASS_A)
+#define CONFIG_LORAMAC_DEFAULT_DEVICE_CLASS     (LORAMAC_CLASS_A)
+#elif IS_ACTIVE(CONFIG_LORAMAC_DEFAULT_DEVICE_CLASS_B)
+#define CONFIG_LORAMAC_DEFAULT_DEVICE_CLASS     (LORAMAC_CLASS_B)
+#elif IS_ACTIVE(CONFIG_LORAMAC_DEFAULT_DEVICE_CLASS_C)
+#define CONFIG_LORAMAC_DEFAULT_DEVICE_CLASS     (LORAMAC_CLASS_C)
+#endif
+
+#ifndef CONFIG_LORAMAC_DEFAULT_DEVICE_CLASS
+#define CONFIG_LORAMAC_DEFAULT_DEVICE_CLASS     (LORAMAC_CLASS_A)
 #endif
 
 /**
  * @brief   Default NetID (only valid with ABP join procedure)
+ *
+ * NETID is used by networks for assigning network-specific addresses to their
+ * end-devices (i.e., DevAddr) so that uplink frames sent by those devices even
+ * when they are roaming outside their home network can be forwarded to their
+ * home network.
  */
-#ifndef LORAMAC_DEFAULT_NETID
-#define LORAMAC_DEFAULT_NETID                  (1U)
+#ifndef CONFIG_LORAMAC_DEFAULT_NETID
+#define CONFIG_LORAMAC_DEFAULT_NETID            (1U)
 #endif
 
 /**
  * @brief   Default network type (public or private)
+ *
+ * This configuration sets the sync word for LoRaWAN communication. This should
+ * be in concordance with the gateway configuration. Public networks use `0x34`
+ * while private networks use `0x12` as sync word.
  */
-#ifndef LORAMAC_DEFAULT_PUBLIC_NETWORK
-#define LORAMAC_DEFAULT_PUBLIC_NETWORK         (true)
-#endif
-/**
- * @brief   Default datarate
- */
-#ifndef LORAMAC_DEFAULT_DR
-#define LORAMAC_DEFAULT_DR                     (LORAMAC_DR_0)
+#ifdef DOXYGEN
+#define CONFIG_LORAMAC_DEFAULT_PUBLIC_NETWORK
```

Comment:
Chances for a person to use a public network is more than that of a private network, hence the default behaviour makes sense. But in that case, I will have to deprecate the current macro since it's `TRUE` by default. Is that fine?

Reply:
Yes